### DISCUSSION TITLE
"The decryption was successfully cracked"

### DISCUSSION THREAD

**Discussion by oakeley** [maheini/FreeStyle-Libre-3-patch#49] [Q&A]
Hi
Is there a code example anywhere that is not dependent on Abbott binaries (in a Jugglucco wrapper) to read and decrypt the Libre3 data or to manage the BLE exchange between a specific sensor and a phone/watch?
Can someone point me to the correct GitHub or GitHub/file to inspect the code?
Thanks

**Comment by twinko** [comment]
no, the reverse engineered is not public

**Comment by klaaslusterbroodvan** [comment]
I have looked for this too. The only information would be to just read the source code of the Juggluco app really carefully, but it is not easy as it seems to be written in a mix of native Android (C) and some Java. I suppose, though, that you can probably start with the Juggluco code and adapt it into a new project without having to fully understand the exact coding underlying the bluetooth or NFC protocols used by the Libre3 app. Anyway, I have a lot of respect for the Juggluco coder who must have spent a tremendous amount of time to make the app work so well.

**Reply by twinko** [reply]
Maybe im wrong but i read somewhere that parts of the code needed to read the sensor is located in external libraries you are not able to read through. So reverse engineering the juggluco app wouldn't help.

**Reply by oakeley** [reply]
Juggluco doesn't have a solution it's just static binaries with a wrapper
